File: apps/dsa-web/src/api/logs.ts
import apiClient from './index';

export type OperationLogItem = {
  id: number;
  category: string;
  action: string;
  level: string;
  status: string;
  title: string;
  message: string;
  stockCode?: string | null;
  stockName?: string | null;
  taskId?: string | null;
  details?: Record<string, unknown> | null;
  createdAt?: string | null;
};

export type OperationLogPage = {
  items: OperationLogItem[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
};

export const logsApi = {
  async getLogs(params?: {
    page?: number;
    pageSize?: number;
    category?: string;
    status?: string;
  }): Promise<OperationLogPage> {
    const response = await apiClient.get<{
      items?: Array<{
        id: number;
        category: string;
        action: string;
        level: string;
        status: string;
        title: string;
        message: string;
        stock_code?: string | null;
        stock_name?: string | null;
        task_id?: string | null;
        details?: Record<string, unknown> | null;
        created_at?: string | null;
      }>;
      total?: number;
      page?: number;
      page_size?: number;
      total_pages?: number;
    }>('/api/v1/logs', {
      params: {
        page: params?.page,
        page_size: params?.pageSize,
        category: params?.category,
        status: params?.status,
      },
    });

    return {
      items: (response.data.items ?? []).map((item) => ({
        id: item.id,
        category: item.category,
        action: item.action,
        level: item.level,
        status: item.status,
        title: item.title,
        message: item.message,
        stockCode: item.stock_code,
        stockName: item.stock_name,
        taskId: item.task_id,
        details: item.details,
        createdAt: item.created_at,
      })),
      total: response.data.total ?? 0,
      page: response.data.page ?? 1,
      pageSize: response.data.page_size ?? 20,
      totalPages: response.data.total_pages ?? 1,
    };
  },
};

File: api/v1/schemas/logs.py
# -*- coding: utf-8 -*-
"""Operation log schemas."""

from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


class OperationLogItem(BaseModel):
    """单条操作日志。"""

    id: int = Field(..., description="日志 ID")
    category: str = Field(..., description="日志分类")
    action: str = Field(..., description="操作类型")
    level: str = Field(..., description="日志级别")
    status: str = Field(..., description="操作状态")
    title: str = Field(..., description="日志标题")
    message: str = Field(..., description="日志内容")
    stock_code: Optional[str] = Field(None, description="股票代码")
    stock_name: Optional[str] = Field(None, description="股票名称")
    task_id: Optional[str] = Field(None, description="关联任务 ID")
    details: Optional[Dict[str, Any]] = Field(None, description="结构化明细")
    created_at: Optional[str] = Field(None, description="创建时间")


class OperationLogListResponse(BaseModel):
    """操作日志列表响应。"""

    items: List[OperationLogItem] = Field(default_factory=list, description="日志列表")
    total: int = Field(0, description="符合筛选条件的总记录数")
    page: int = Field(1, description="当前页码")
    page_size: int = Field(20, description="每页条数")
    total_pages: int = Field(1, description="总页数")
